### PR TITLE
Update custom recordClass behavior to allow for multiple data stores.

### PR DESCRIPTION
**Pull request by kvandelden**
Update custom recordClass behavior to allow for multiple data stores.
Mimic Record behavior of extending class.

### CODE REVIEW

**Comment by kvandelden** [comment]
Needs to be revised.. don't review..

**Comment by kvandelden** [comment]
The solution works for es5 functions, however, if you have an es6 class, you receive the error.
TypeError: Class constructor CustomRow cannot be invoked without 'new'.

**Comment by kvandelden** [comment]
This is now ready for Review..

Used the following as a reference to get es5 code to extend an es6 class.
https://esdiscuss.org/topic/extending-an-es6-class-using-es5-syntax

Thanks, K

**Comment by jmdobry** [comment]
Can you explain why this PR is necessary?

**Comment by kvandelden** [comment]
This addresses issue  #466

**Comment by jmdobry** [comment]
You mean #465?

**Comment by kvandelden** [comment]
Yes, you are correct..  my mistake

File: src/Mapper.js [hunk 1]
```diff
@@ -480,8 +480,18 @@ function Mapper (opts) {
         return subClass
       })()
     })
+  } else if (this.recordClass !== false) {
+    const customSuperClass = this.recordClass
+    this.recordClass = function newRecordClass (props, opts) {
+      var base = Reflect.construct(customSuperClass, arguments, newRecordClass); // eslint-disable-line
```
Comment (jmdobry) [inline src/Mapper.js]: s/var/const/

Comment (jmdobry) [inline src/Mapper.js]: How does `Reflect.construct` transpile to ES5? I've never used it before.

**Comment by crobinson42** [comment]
@kvandelden any interest in moving this along with the requested changes?

File: src/Mapper.js (hunk 1 above)
Comment (kvandelden) [reply src/Mapper.js]: I reviewed our webpack output, and it appears it doesn't appear to convert it to es5. 

I am not familiar enough with this part of js to know a path forward with ES5.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Reflect/construct

**Comment by crobinson42** [comment]
@kvandelden do you want to write some tests for this so we can understand the use case and benefit as it might apply to #465 ?

**Comment by crobinson42** [comment]
closing - lack of activity

**Pull request closed without merging**